### DISCUSSION TITLE
Is it possible to have a file based registry AND pull it from github?

### DISCUSSION THREAD

**Discussion by diehard2** [microsoft/vcpkg#30428] [General]
I'm making a registry for some internal libraries. I went with file based because its far easier to understand the structure and method for updating for other developers. However, I don't seem to be able to pull it in from other repositories with a vcpkg-configuration.json setup. It can find the repo, but can't parse the file structure. Is there anyway to pull this off? I could switch to the git versioning, but honestly that's really hard to grasp in comparison and I don't want to add any more friction for people who are not well versed in vcpkg updating packages.

error: failed to parse versions file /home/user/sdanauskas/.cache/vcpkg/registries/git-trees/606151ff8a11a622ffcbacc57dfec7b855e619b6/f-/fdsrpc.json$.versions[0] (a version database entry): unexpected field 'path', did you mean 'version'?

**Comment by autoantwort** [comment]
Have you set `"kind": "filesystem"`? I guess you can make use of the filesystem registry by using a git submodule

**Reply by autoantwort** [reply]
Also using overlay ports is maybe enough depending of your use case

**Reply by diehard2** [reply]
Thanks, would be nice if this worked since its much easier for developers, but I can make it a submodule.
